lib/mcb/commands/users/revoke.rb

```diff
@@ -1,13 +1,19 @@
 summary 'Remove a user from an organisation/provider in the DB.'
 param :id_or_email_or_sign_in_id
-param :provider_code, transform: ->(code) { code.upcase }
-usage 'revoke <user_id/email/sign_in_user_id> <provider_code>'
+option :p, 'provider_code', 'provider code',
+       argument: :optional, transform: ->(code) { code.upcase }
+usage 'revoke <user_id/email/sign_in_user_id> [--provider-code <provider_code>]'
 
 run do |opts, args, _cmd|
   MCB.init_rails(opts)
 
   cli = HighLine.new
-  provider = Provider.find_by!(provider_code: args[:provider_code])
+
+  provider = nil
+  if opts[:provider_code]
+    provider = Provider.find_by!(provider_code: opts[:provider_code])
+  end
```

Comment:
More golfing ... I think that this would work as well, and be shorter:

```
    provider = Provider.find_by(provider_code: opts[:provider_code])
```

Basically, just allow `find_by` to return a nil, which is what you'll get if you run `find_by(nil)`. Embrace the `nil`!!

Reply:
much tidier. I hadn't understood that it was just the bang making it go bang!

all done, rebased & pushed, cheers for those